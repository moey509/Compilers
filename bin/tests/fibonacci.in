class Fib {
	fun recursiveFib(int: i){
		if (i < 3) {
			return 1;
		}
		else {
			return recursiveFib(i-1) + recursiveFib(i-2);
		}
	}
	
	fun iterativeFib(int: i){
		if (i <= 2) {
			return 1;
		}
		else {
			int temp1 = 1;
			int temp2 = 1;
			int counter = 2;
			while (counter < i){
				counter++;
				temp2 = temp1 + temp2;
				temp1 = temp2 - temp1;
			}
			return temp2;
		}		
	}			
}