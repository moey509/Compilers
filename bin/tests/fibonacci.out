class Name { fun ( int : name  ) { if ( name < 0 ) { return 0 ; } else { return name ( name - 0 ) + name ( name - 0 ) ; } } fun name ( int : name ) { if ( name <= 0 ) { return 0 ; } else { int name = 0 ; int name = 0 ; int name = 0 ; while ( name < name ) { name ++ ; name = name + name ; name = name - name ; } return name ; } } }